tripal/tripal
# Drush command to create Tripal + Chado Fields

# Tripal 4 Core Dev Task

**DEPENDANT ON PR #1612**

Issue #1606


### Tripal Version: 4

## Description


This PR adds 8 Drush generators for Tripal/Chado Fields. Specifically, it adds

```
  tripal:field            Generates a Tripal Field Type, Widget and Formatter for fields not interacting with Chado.
  tripal:field-formatter  Generates a Tripal Formatter to be used with an existing Tripal Field.
  tripal:field-type       Generates a Tripal Field Type for developing Tripal fields with no interactiion with Chado.
  tripal:field-widget     Generates a Tripal Widget to be used with an existing Tripal Field.
  tripal-chado:field            Generates a Chado Field Type, Widget and Formatter for fields who store their data in Chado.
  tripal-chado:field-formatter  Generates a Chado Formatter to be used with an existing Chado Field.
  tripal-chado:field-type       Generates a Chado Field Type for developing fields which store their data in chado.
  tripal-chado:field-widget     Generates a Chado Field Widget to be used with an existing Chado Field.
```

Once this PR is merged, any developer can generate the skeletons of these field types by entering `drush generate ` followed by one of the generators listed above.  Then they enter the prompts and the generator will generate the files using the values they put in.

## Testing?



1. Switch to this branch on a fresh Tripal clone.
2. Rebuild the cache using `drush cr`
3. Execute `drush generate tripal` and confirm you see 4 Tripal field generators.
4. Execute `drush generate tripal-chado` and confirm you see 4 Chado field generators.
5. Test each of the generators by running the command, entering the criteria and then testing the generated field.

NOTE: The Tripal generator default generate a completely functioning field. However, the Chado field generators do not... to test these fully you will have to add at least one property to the generated field type class. This was done because there wasn't a simple example to generate that did not cause unexpected results.
